### PR TITLE
RS/YJ/Rule 1-3

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
~~This rule hasn't been tested yet. Thanks!~~

This rule has been tested yet. Thanks!

### CODE REVIEW

**Comment by Jiarongx-Xie** [comment]
This looks good to me. Thank you for the nice work!

File: rct229/rulesets/ashrae9012019/section1/section1rule3.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        bpf_set = calc_vals["bpf_set"]
+        bbp_set = calc_vals["bbp_set"]
+        bbrec_set = calc_vals["bbrec_set"]
+        bbuec_set = calc_vals["bbuec_set"]
+
+        return len(pci_target_set) == len(bpf_set) == len(bbp_set) == len(
+            bbrec_set
+        ) == len(bbuec_set) == 1 and std_equal(
+            (bbuec_set[0] + (bpf_set[0] * bbrec_set[0]) / bbp_set[0]), pci_target_set[0]
+        )  # TODO: what if  bbp_set[0] == 0?
```
Comment (JacksonJ-KC) [inline rct229/rulesets/ashrae9012019/section1/section1rule3.py:115]: Same recommendation as for rule 1-2 where we ensure there is not division by 0 by adding a condition that comes before the division, then adding a fail message

Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section1/section1rule3.py]: Addressed

**Review by JacksonJ-KC — APPROVED** [review]

**Review by Jiarongx-Xie — APPROVED** [review]

File: rct229/rulesets/ashrae9012019/section1/section1rule3.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
+            rmrs_used_optional=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section1/section1rule3.py:36]: Remove user, baseline_0 and proposed or set them to False.

File: rct229/rulesets/ashrae9012019/section1/section1rule3.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                )
+                bbp_set.append(
+                    find_one("$.output.baseline_building_performance_energy_cost", rmd)
+                )
+                bbrec_set.append(
+                    find_one("$.output.baseline_building_regulated_energy_cost", rmd)
+                )
+                bbuec_set.append(
+                    find_one("$.output.baseline_building_unregulated_energy_cost", rmd)
+                )
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section1/section1rule3.py:79]: After the `for` loop, you may need to add an assertion to make sure those lists are not empty.

```python
assert_(pci_target_set and bpf_set and bbp_set and bbrec_set and bbuec_set, "Required data is missing in the RDS, make sure all RDSs have the following data: output.performance_cost_index_target, ....")
```

File: rct229/rulesets/ashrae9012019/section1/section1rule3.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+        pci_target_set = filter_(pci_target_set, lambda x: x is not None)
+        bpf_set = filter_(bpf_set, lambda x: x is not None)
+        bbp_set = filter_(bbp_set, lambda x: x is not None)
+        bbrec_set = filter_(bbrec_set, lambda x: x is not None)
+        bbuec_set = filter_(bbuec_set, lambda x: x is not None)
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section1/section1rule3.py:85]: Use the native filter function

**Review by weilixu — APPROVED** [review]
Approved with some comments need to address.

File: rct229/rulesets/ashrae9012019/section1/section1rule3.py (hunk 4 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section1/section1rule3.py:85]: Addressed.

File: rct229/rulesets/ashrae9012019/section1/section1rule3.py (hunk 3 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section1/section1rule3.py:79]: The assert_ messages are available in lines 86-93.

File: rct229/rulesets/ashrae9012019/section1/section1rule3.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section1/section1rule3.py:36]: I tested this out and decided to leave this as is b/c ```USER```, ```BASELINE_0```, and ```PROPOSED``` are also optional for this ruleset.

**Pull request merged**